### ISSUE TITLE
Issue with polish "Ł"

### ISSUE THREAD

**Issue by bigretromike** [sumatrapdfreader/sumatrapdf#2150]
I got invoice with "zł" as payment sign. in 3.3.3 there is square in place, on pre-3.4.14080 there is empty space.

**Comment by GitHubRulesOK** [comment]
Sample needed 
BUT do not post personal data files

No problem with other files just the common issues over poor PDFs with poor Unicode font mapping
in england it could be £ or € in Denmark see https://github.com/sumatrapdfreader/sumatrapdf/issues/2069


![image](https://user-images.githubusercontent.com/12754913/131324461-cdf22f55-d133-441f-a080-abb8edae3fb9.png)

**Comment by bigretromike** [comment]
I did sent "sample" to @kjk any clue why it stopped to work ?

**Comment by kjk** [comment]
I looked at it but don't really know. 

The file is not generated correctly. Apparently earlier versions of mupdf was able to repair the file but latest version cannot. It's not easy to tell why because a lot of code has changes and it's complicated code.

**Comment by bigretromike** [comment]
happy that sample arrived to you. 
Yes something did change in mupdf, and that thing prevent us from upgrading pdfsumatra on-site :(

**Comment by GitHubRulesOK** [comment]
The core changes in MuPDF since 3.1.2 are mainly down to two key changes, 
one is greater dependency on not correcting included font library with mainly some issues over white space (n/a in this case) and determining font substitution including unicode (may or may not be relevant)
two is stricter MuPDF rejection / lesser tolerance of malformed font definitions (more likely) Those can sometimes be corrected via a bug report at MuPDF.
The simple solution as shown in my example is when creating a PDF use correctly defined fonts that natively support embedding European characters.

**Comment by bigretromike** [comment]
> 
> 
> The simple solution as shown in my example is when creating a PDF use correctly defined fonts that natively support embedding European characters.

That wouldn't be a problem if those PDF files was create by me, we get those from outside source, and currently because of issue with displaying printing also is missing characters.

While we can stay on old sumatra version for some time, we will need to migrate to something else... because more and more people demand crypto support (validation of sign) and the issue with characters is also annoying.

But thanks for you sugestion, @GitHubRulesOK

**Comment by GitHubRulesOK** [comment]
Try to determine if the fault is due to one source client/software 
then find an equivalent sample that is not too complex (single page) 
or if private test **identical fault** after very simple deletions of names in an editor
Then the sample may be suitable for bug reporting to MuPDF or send simplified to kjk

**Comment by bigretromike** [comment]
> 
> 
> Try to determine if the fault is due to one source client/software
> then find an equivalent sample that is not too complex (single page)
> or if private test **identical fault** after very simple deletions of names in an editor
> Then the sample may be suitable for bug reporting to MuPDF or send simplified to kjk

I did sent sample to kjk

**Comment by kjk** [comment]
Closing: not enough information. This is almost always a PDF font-embedding issue (missing glyphs for characters like "Ł"/"zł"), and no sample PDF was provided after requests for one.

If you still see this on a current pre-release, reopen with a minimal sample PDF that shows the problem.

**Comment by bigretromike** [comment]
yeah - maybe if it was asked 5 years ago 👍 i could provide that file on gh
I did sent some samples cant remember if it was via mail or something. 

Sorry

I dont use sumatrapdf as much as I did then, lack of (back then) ability to check proper digital signature killed the usage in government facility, while being super lightweight which was super helpful back then, the lack of some features and issues with printing from it (missing layers) was to much - had to move more and more people to Adobe and I moved on.
